mesonbuild/dependencies/pkgconfig.py

```diff
@@ -31,19 +31,46 @@
     from ..utils.core import EnvironOrDict
     from .._typing import ImmutableListProtocol
 
+
+def get_env(environment: Environment, for_machine: MachineChoice,
+            uninstalled: bool = False) -> EnvironmentVariables:
+    env = EnvironmentVariables()
+    key = OptionKey('pkg_config_path', machine=for_machine)
+    extra_paths: T.List[str] = environment.coredata.options[key].value[:]
+    if uninstalled:
+        uninstalled_path = Path(environment.get_build_dir(), 'meson-uninstalled').as_posix()
+        if uninstalled_path not in extra_paths:
+            extra_paths.append(uninstalled_path)
+    env.set('PKG_CONFIG_PATH', extra_paths)
+    sysroot = environment.properties[for_machine].get_sys_root()
+    if sysroot:
+        env.set('PKG_CONFIG_SYSROOT_DIR', [sysroot])
+    pkg_config_libdir_prop = environment.properties[for_machine].get_pkg_config_libdir()
+    if pkg_config_libdir_prop:
+        env.set('PKG_CONFIG_LIBDIR', pkg_config_libdir_prop)
+    return env
+
+
 class PkgConfigInterface:
     '''Base class wrapping a pkg-config implementation'''
 
     @staticmethod
     def instance(env: Environment, for_machine: MachineChoice, silent: bool) -> T.Optional[PkgConfigInterface]:
-        impl = PkgConfigCLI(env, for_machine, silent)
+        impl = PkgConfigInterface()
```

Comment:
Why? It tells mypy the type of the variable, and it acts as a stub where found() is always False, until a right implementation is found.

Reply:
`impl: PkgConfigInterface` is valid syntax. You don't need to construct it.

Reply:
The type is secondary here. What I need is a not found PkgConfigInterface. Otherwise, I would need to add a local bool variable to track whether I found a valid implementation or not.

If there is no entry for pkg-config in the machine files, I try PyPkgConf, and it may be found or not. If there is an entry, then the impl is never found at this point, and I try PkgConfigCLI.

Reply:
```python
impl: T.Optional[PkgConfigInterface] = None

if all(env.lookup_binary_entry(for_machine, potential_name) is None for potential_name in env.default_pkgconfig):
   impl = PyPkgConf(env, for_machine, silent)

if not impl or not impl.found():
  impl = PkgConfigCLI(env, for_machine, silent)

if not impl.found():
   return None

return impl
```

Why does this not work?

Reply:
It works. I just find it more complicated...